certbot/tests/compat/filesystem_test.py

```diff
@@ -6,41 +6,162 @@
 from certbot.tests.util import TempDirTestCase
 
 
-class FilesystemTest(TempDirTestCase):
-    """Unit tests for filesystem module."""
-    @unittest.skipIf(os.name != 'nt', reason='Test specific to Windows security')
+@unittest.skipIf(os.name != 'nt', reason='Test specific to Windows security')
+class WindowsChmodTests(TempDirTestCase):
+    """Unit tests for Windows chmod function in filesystem module"""
+    def test_symlink_resolution(self):
+        probe_path = _create_probe(self.tempdir)  # This is 0o744 by default
```

Comment:
Does the initial mode here not depend on the environment the test its run in? On UNIX it depends on the umask. On Windows, I'd expect it to depend on the parent directories.

Reply:
You are right. I remove the comment.

Reply:
If the default permissions do depend on the parent directories (and that remains true all the way up the filesystem), isn't it possible this test could fail depending on the configuration of the system its run on?

If so, maybe it's worth calling `filesyste.chmod(tempdir, 0o744)` in `setUp` or `_create_probe` for a little extra stability here?

Reply:
It is worth it. Added to `_create_probe`.